keps/sig-windows/1981-windows-privileged-container-support/README.md

```diff
@@ -452,94 +462,203 @@ change are understandable. This may include API specs (though not always
+
+#### Container users
+
+- The privileged container can run as any user that's available on the host or in the domain of the host machine. Password accounts are being investigated.
+Running privileged containers as non SYSTEM/admin accounts will be the primary way operators can restrict access to system resources (files, registry, named pipes, WMI, etc).
+More information on Windows resource access can be found at https://docs.microsoft.com/en-us/archive/msdn-magazine/2008/november/access-control-understanding-windows-file-and-registry-permissions.
+
 #### Container Mounts
-- Directory mounts (i.e. directory C:\test mapped to C:\test in the privileged container) are still being investigated. Note that file system mapping in the current implementation would expose both the host filesystem and the container filesystem to the processes in the privileged container. We are still investigation how external mounts such as secrets or storage accounts would be added. The default visibility of all the host filesystem may present backwards compatibility challenges in the future if host path mounting is enabled in Windows. Currently the only method to restrict would be via the user in a way similar to [UID or GID](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) in the PSP. Currently however, this host path mounting is unlikely to be enabled in the future as a restriction of Job Objects.  
+
+- When privileged containers are started a new volume will be created on the host which will contain the contents of the container image. Containers will have a default working directory that points to this container volume. Containers will also have full access to the host file-system (unless restricted by filed-based ACLs and the run_as_username used to start the container.) Processes should use absolute paths when accessing files on the host and relative paths when accessing files brought in via the container image.
+Note: there will be no `chroot` equivalent.
+- An environment variable `$CONTAINER_SANDBOX_MOUNT_POINT` will be set to the absolute path where the container volume is mounted.
+- Volume mounts (including service account tokens) will be supported for privileged containers and will be mounted under the container volume. Programs running inside the container can either access volume mounts be using a relative path or by prefixing `$CONTAINER_SANDBOX_MOUNT_POINT` to their paths (example: use either `.\var\run\secrets\kubernetes.io\serviceaccount\` or `$CONTAINER_SANDBOX_MOUNT_POINT\var\run\secrets\kubernetes.io\serviceaccount\` to access service account tokens). These relative paths will be based on `Pod.containers.volumeMounts.mountPath`.
+- Client libraries such as https://pkg.go.dev/k8s.io/client-go/rest#InClusterConfig will be updated to prefix paths with `$CONTAINER_SANDBOX_MOUNT_POINT` if the environment variable is set for Windows so these libraries will work in `hostProcess`containers.
```

Comment:
a few points on this:
* this isn't possible to feature-gate
* there are a lot of clients (client-go and https://github.com/kubernetes-client are the ones the kubernetes project ships)
* this will take a while to propagate out to client consumers

It would be helpful to describe when this is planned (I would guess not until beta, so we don't leak alpha behavior into clients?) and to add sig-api-machinery as participating since they oversee client libraries

Reply:
Sorry, I missed the part about client library prefixing paths. Which paths are you referring to here?

Reply:
@msau42 the client library prefxing path updates would be added to the client libraries to help programs running inside these privileged containers find files in well-known mounts such as the service-account token.

@liggitt I agree beta would be a good time to do this, I'll list that as a beta graduation criteria. For alpha applications can rely on relative paths

Reply:
Programs today do not need client-go linked in to access service account tokens. The purpose of volumeMounts is so we put the files where the containers expect them to be.

Reply:
@msau42 understood. 
On normal windows contianers today `/var/run/secrets` gets mounted to `c:\var\run\secrets\`.
With these privileged container `/var/run/secrets` will get mounted to a directory under $CONTAINER_SANDBOX_MOUNT_POINT (today c:\c\{containerid}).
Unfortunately there is no way around this (if we mapped it to `c:\var\run\secrets\` they would collide
The updates to the client libraries will be it make it easier to find the paths.

Reply:
@liggitt i proposed discussing updating the client libraries until after alpha so we can get more feedback.